tests/lax_control_flow_test.py

```diff
@@ -999,6 +999,24 @@ def f(x):
       self.assertAllClose(ans, expected, check_dtypes=False)
       jtu.check_grads(f, (x,), order=2, modes=["fwd", "rev"])
 
+  def testSwitchGradWithWeakTypeMismatch(self):  # issue 4696
+    branches = [
+        lambda x: x,           # This preserves the weak type of x.
+        lambda x: jnp.cos(x),  # This strips the weak type of x.
```

Comment:
In the near future unary operations will likely preserve weak types. To future proof this, you could do something like `lambda x: x + jnp.float32(1)`